### PR TITLE
Add web workflow to default validators

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

The WebWorkflow was missing from the doctor validators. This would prevent a user that has an available "web device" from running flutter if they didn't have any other functioning devices.

Discovered by @natebosch

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/commands/doctor_test.dart [hunk 1]
```diff
@@ -845,3 +859,4 @@ class VsCodeValidatorTestTargets extends VsCodeValidator {
 }
 
 class MockProcessManager extends Mock implements ProcessManager {}
+class MockPlatform extends Mock implements Platform {}
```
Comment (natebosch) [inline packages/flutter_tools/test/general.shard/commands/doctor_test.dart]: Is this used?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/commands/doctor_test.dart]: Nope, fixed

File: packages/flutter_tools/lib/src/doctor.dart [hunk 2]
```diff
@@ -113,6 +113,9 @@ class _DefaultDoctorValidatorsProvider implements DoctorValidatorsProvider {
       if (windowsWorkflow.appliesToHostPlatform)
         _workflows.add(windowsWorkflow);
 
+      if (webWorkflow.appliesToHostPlatform)
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/doctor.dart]: (optional, nit) switch to new syntax.
```dart
return _workflows ??= <Workflow>[
  if (iosWorkflow.appliesToHostPlatform) iosWorkflow,
  if (androidWorkflow.appliesToHostPlatform) androidWorkflow,
  ...
];
```

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/lib/src/doctor.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/doctor.dart]: Worth doing, but there are some timing constraints with this fix...

**Pull request merged**